purchase_request/models/stock_move.py

```diff
@@ -108,41 +108,43 @@ def copy_data(self, default=None):
         """
         if default is None:
             default = {}
-        if not default.get("purchase_request_allocation_ids") and (
-            default.get("product_uom_qty") or self.state in ("done", "cancel")
-        ):
-            default["purchase_request_allocation_ids"] = []
-            new_move_qty = default.get("product_uom_qty") or self.product_uom_qty
-            rounding = self.product_id.uom_id.rounding
-            for alloc in self.purchase_request_allocation_ids.filtered(
-                "open_product_qty"
+        vals_list = super().copy_data(default)
+        for move, vals in zip(self, vals_list, strict=False):
+            if not default.get("purchase_request_allocation_ids") and (
+                default.get("product_uom_qty") or move.state in ("done", "cancel")
             ):
-                if (
-                    float_compare(
-                        new_move_qty,
-                        0,
-                        precision_rounding=self.product_id.uom_id.rounding,
-                    )
-                    <= 0
-                    or float_compare(
-                        alloc.open_product_qty, 0, precision_rounding=rounding
-                    )
-                    <= 0
+                vals["purchase_request_allocation_ids"] = []
```

Comment:
In the previous situation it was `default` that was populated with the allocations, meaning that if a value for `purchase_request_allocation_ids` was passed, this mechanism would be circumvented. Because you set up the allocations directly in `vals`, this is no longer possible. You might want to check if a value for `purchase_request_allocation_ids` is not already present in vals and do nothing in that case.

Reply:
Apologies, that is not how the `default` argument works. I was mislead by its name!

Reply:
Sorry, you already check this. Ignore.